crates/ty_python_semantic/src/types/signatures.rs

```diff
@@ -1245,10 +1254,22 @@ impl<'db> Parameters<'db> {
                 && !is_classmethod
                 && arg.parameter.annotation().is_none()
                 && parameters.index(arg.name().id()) == Some(0)
+                && let Some(class) = class_type
+                && let Some(method) = method_type
             {
-                let implicit_annotation = Type::SpecialForm(SpecialFormType::TypingSelf)
-                    .in_type_expression(db, definition.scope(db), Some(definition))
-                    .ok();
+                let is_method_generic = method
+                    .signature(db)
+                    .overloads
+                    .iter()
+                    .any(|s| s.generic_context.is_some());
+                let is_class_generic = !class.is_not_generic();
```

Comment:
Do we need to even check whether the class is generic? Does it cause any bad result if we don't?